### PR TITLE
fix: Inline links

### PR DESCRIPTION
**Pull request by flochtililoch**
When rendering `hyper-ref` component within a text element, use `Text` component instead of `TouchableOpacity`.

Note: I tried using more modern `Pressable` component as a replacement of `TouchableOpacity`, but it did not allow keeping inline layout on nested `<text>`.

Solves #307

https://user-images.githubusercontent.com/309515/180471486-f5ba11fa-d502-4eed-8cb9-b1d59e0a5d9b.mp4

https://user-images.githubusercontent.com/309515/183530120-271e4c5d-dac2-4953-a000-f73dd5dd2f3d.mov

### CODE REVIEW

File: src/core/hyper-ref/index.js [hunk 1]
```diff
@@ -305,10 +304,51 @@ export default class HyperRef extends PureComponent<Props, State> {
...
+      this.props.element.parentNode?.localName === LOCAL_NAME.TEXT;
+
+    if (isNestedUnderText) {
+      const noop = () => {};
+      return (
+        <Text
+          accessibilityLabel={accessibilityLabel}
+          accessible={false}
+          onLongPress={onLongPress}
+          // when no press handler set, we still need an empty handler for pressIn or pressOut handlers to work
```
Comment (adamstep) [inline src/core/hyper-ref/index.js:323]: How did you figure out this issue and the one below for `onResponderRelease`/`onResponderRelease`? Is there an SO of GH issue we can link to?

File: src/core/hyper-ref/index.js [hunk 2]
```diff
@@ -305,10 +304,51 @@ export default class HyperRef extends PureComponent<Props, State> {
...
+          onPress={onPress || (onPressIn || onPressOut ? noop : undefined)}
```
Comment (flochtililoch) [inline src/core/hyper-ref/index.js:324]:
```suggestion
          onPress={onPress || ((onPressIn || onPressOut) ? noop : undefined)}
```

File: src/core/hyper-ref/index.js (hunk 1 above)
Comment (flochtililoch) [reply src/core/hyper-ref/index.js:323]: I just ran into the bug myself, I can check if there's any document out there for this

File: src/core/hyper-ref/index.js (hunk 2 above)
Comment (flochtililoch) [reply src/core/hyper-ref/index.js:324]: hmm I guess prettier does not want these extra parenthesis and `?` is interpreted on the whole expression, not just the last bit, so change is not needed.

File: src/core/hyper-ref/index.js (hunk 1 above)
Comment (flochtililoch) [reply src/core/hyper-ref/index.js:323]: I haven't found anything of value in GH/SO so far. I might pause this one a little bit and do a bit more testing on it including Android

**Review by adamstep — APPROVED** [review]

**Review by abdelav — APPROVED** [review]

File: src/core/hyper-ref/index.js (hunk 1 above)
Comment (flochtililoch) [reply src/core/hyper-ref/index.js:323]: Tested on Android 12, it's working well.

**Pull request merged**